iidm/iidm-converter-api/src/main/java/eu/itesla_project/iidm/datasource/ReadOnlyMemDataSource.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2016, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.iidm.datasource;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Giovanni Ferrari <giovanni.ferrari@techrain.it>
+ */
+public class ReadOnlyMemDataSource implements ReadOnlyDataSource {
+
+    static class Key {
+
+        private String suffix;
+
+        private String ext;
+
+        Key(String suffix, String ext) {
+            this.suffix = suffix;
+            this.ext = ext;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(suffix, ext);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj instanceof Key) {
+                Key other = (Key) obj;
+                return Objects.equals(suffix, other.suffix)
+                        && Objects.equals(ext, other.ext);
+            }
+            return false;
+        }
+
+        @Override
+        public String toString() {
+            return Objects.toString(suffix) + "." + Objects.toString(ext);
+        }
+
+    }
+
+    protected final Map<Key, byte[]> data = new HashMap<>();
+
+    protected final Map<String, byte[]> data2 = new HashMap<>();
```

Comment:
Maybe we can have only one Map and create "filename" from a basename, a suffix and an extension